File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        final = []
        for i in range(len(nums)):
            temp = nums.copy()
            print(i)
            temp.pop(i)
            print(temp)
            product = 1
            for num in temp:
                product *= num
            final.append(product)
        return final

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        set<int> m1;

        for (int i = 0; i<nums.size(); i++){
            auto it = m1.find(nums[i]);
            if (it != m1.end()){
                return true;
            }
            m1.insert(nums[i]);
        }
        return false;
    }
};